formats/hocon/src/test/kotlin/kotlinx/serialization/hocon/HoconEncoderTest.kt

```diff
@@ -0,0 +1,118 @@
+package kotlinx.serialization.hocon
+
+import kotlinx.serialization.*
+import org.junit.*
+
+class HoconEncoderTest {
+
+    @Serializable
+    data class SimpleConfig(val value: Int)
+
+    @Serializable
+    data class PrimitivesConfig(
+        val b: Boolean,
+        val i: Int,
+        val d: Double,
+        val c: Char,
+        val s: String,
+        val n: String?,
+    )
+
+    @Test
+    fun encodeSimpleConfig() {
+        val obj = PrimitivesConfig(b = true, i = 42, d = 32.2, c = 'x', s = "string", n = null)
+        val config = Hocon.encodeToConfig(obj)
+
+        config.assertContains("b = true, i = 42, d = 32.2, c = x, s = string, n = null")
+    }
+
+    @Serializable
+    data class ConfigWithEnum(val e: RegularEnum)
+
+    @Serializable
+    enum class RegularEnum { VALUE }
+
+    @Test
+    fun encodeConfigWithEnum() {
+        val obj = ConfigWithEnum(RegularEnum.VALUE)
+        val config = Hocon.encodeToConfig(obj)
+
+        config.assertContains("e = VALUE")
+    }
+
+    @Serializable
+    class ConfigWithIterables(
```

Comment:
Could you please also add a test for the class that contains a set or a list of other classes with nullability?

E,g,
```
class Foo(
val set: Set<Bar?>?
)
```

with both nulls and non-nulls as container and as its elements when its not null